### PR TITLE
Copy and save logs from deployment pane

### LINKED ISSUE #2076: As a scientific researcher I want to copy and download serial logs so that I can analyze my sensor data in spreadsheets
When collecting data from sensors for some scientific experiment it is often desirable to get a time-series table for further analysis in Excel or another external tool.

I can use `console-log` and then manually:

- Scroll the output to the top
- Hold down the mouse button to start the selection
- Scroll down to select all
- Press Ctrl+C to copy
- Paste elsewhere

But repeating the process again and again is tiresome. We could simplify researches by providing simple accessible buttons to: 

- Copy the whole output to the clipboard
- Save the whole output to a file specified

image

Generalizing the idea, the same buttons will be quite helpful on all other tabs as well. For example, to ask questions on the forum or make bug reports.

### User story

- I make a patch with `console-log` to gather some data
- I upload with debug and see the data coming to the Serial tab
- I press the “Copy” button and the contents of the Serial tab are copied to the clipboard exactly as I see them, applying the current filter
- I press the “Save to File” button and the system file save dialog appears asking for the filename (defaults to `serial-log.txt`)
- I enter the filename and the file gets saved with exactly the same contents as I see them

### Acceptance criteria

- [x] The basic scenario works as described
- [x] If I choose an existing file XOD asks me if I want to overwrite it
- [x] If I copy a log made with `join` with TAB delimiter and then paste in Excel the columns get split correctly (the leading arrow in first is OK)
- [x] If I do the same with Google Spreadsheets, the result is split too
- [x] If I choose to include watched values to the log and have some, their log entries are copied/saved too
- [x] The buttons work the same way for all tabs of the deployment pane
- [x] Whenever I press the “Copy” button it gives some visual feedback acknowledging success

### Out of scope

- Filtering out arrows, headers, footers

### PR DESCRIPTION
**Pull request by brusherru**
It closes #2076.

### CODE REVIEW

File: packages/xod-client/src/debugger/messages.js [hunk 1]
```diff
@@ -48,3 +48,26 @@ export const DO_NOT_USE_TETHERING_INTERNET_IN_BROWSER = {
   solution:
     'Install the desktop version of the XOD IDE to reveal all features of XOD',
 };
+
+export const logCopyDone = tabLabel => ({
+  title: `${tabLabel} log is copied to the clipboard`,
```
Comment (nkrkv) [inline packages/xod-client/src/debugger/messages.js:53]:
```suggestion
  title: `Copied`,
```

File: packages/xod-client/src/debugger/messages.js [hunk 2]
```diff
@@ -48,3 +48,26 @@ export const DO_NOT_USE_TETHERING_INTERNET_IN_BROWSER = {
...
+});
+export const logCopyError = (tabLabel, error) => ({
+  title: `Can not copy the ${tabLabel} log`,
```
Comment (nkrkv) [inline packages/xod-client/src/debugger/messages.js:56]:
```suggestion
  title: `Cannot copy log`,
```

File: packages/xod-client/src/debugger/messages.js [hunk 3]
```diff
@@ -48,3 +48,26 @@ export const DO_NOT_USE_TETHERING_INTERNET_IN_BROWSER = {
...
+  note: error.message,
+  solution: 'Please, report the issue to the XOD developers ',
+});
+export const logCopyNotSupported = () => ({
+  title: 'Your browser does not support a copy log feature',
+  solution:
+    'Try to install the desktop IDE or use the latest version of Google Chrome.',
```
Comment (nkrkv) [inline packages/xod-client/src/debugger/messages.js:63]:
```suggestion
  title: 'Bad browser',
  title: 'Your browser does not support a copy log feature',
  solution:
    'Install the desktop IDE or use the latest version of Google Chrome / Firefox.',
```

File: packages/xod-client/src/debugger/messages.js [hunk 4]
```diff
@@ -48,3 +48,26 @@ export const DO_NOT_USE_TETHERING_INTERNET_IN_BROWSER = {
...
+});
+
+export const logSaveDone = tabLabel => ({
+  title: `${tabLabel} log is saved on your filesystem`,
+});
```
Comment (nkrkv) [inline packages/xod-client/src/debugger/messages.js:68]: :bug: In desktop IDE this message appears before the file save dialog pops up, even if I choose to cancel (in the future). If it’s hard to make the message appear in the right moment (after the dialog), it’s OK to not show it at all.

File: packages/xod-client/src/debugger/messages.js [hunk 5]
```diff
@@ -48,3 +48,26 @@ export const DO_NOT_USE_TETHERING_INTERNET_IN_BROWSER = {
...
+export const logSaveError = (tabLabel, error) => ({
+  title: `Can not save the ${tabLabel} log`,
```
Comment (nkrkv) [inline packages/xod-client/src/debugger/messages.js:70]:
```suggestion
  title: `Cannot save log`,
```

**Review by nkrkv — CHANGES_REQUESTED** [review]

File: packages/xod-client/src/debugger/containers/Debugger.jsx [hunk 6]
```diff
@@ -79,11 +138,42 @@ class Debugger extends React.Component {
     );
   }
 
+  renderTabActions() {
+    const isDisabled = R.isEmpty(this.props.log);
+    const title = notDisabledTitle =>
```
Comment (evgenykochetkov) [inline packages/xod-client/src/debugger/containers/Debugger.jsx:143]: I don't think the title should be altered. The reason for disabling the buttons when the log is empty is obvious enough.

**Comment by evgenykochetkov** [comment]
Besides the save message in the desktop IDE, works absolutely fine in both browser and desktop versions 👍🏻

**Review by nkrkv — APPROVED** [review]
:+1:

**Pull request merged**